ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -955,53 +1063,126 @@ namespace OrdinalAxis {
+            // value basing on ordinalPositions
+            if (
+                ordinalPositions[0] <= val &&
+                ordinalPositions[ordinalLength - 1] >= val
+            ) {
+                let index =
+                    Composition.findIndexOf &&
+                    Composition.findIndexOf(ordinalPositions, val, true);
+                // first look for an exact match in the ordinalpositions array
+                if (ordinalPositions[index] === val) {
+                    ordinalIndex = index;
+                } else {
+                    const percent =
+                        (val - ordinalPositions[index]) /
+                        (ordinalPositions[index + 1] - ordinalPositions[index]);
+                    ordinalIndex = index + percent;
+                }
+                // final return value is based on ordinalIndex
             } else {
 
-                let ordinalLength = ordinalPositions.length,
-                    i,
-                    distance,
-                    ordinalIndex;
+                if (!extendedOrdinalPositions) {
+                    extendedOrdinalPositions =
+                        ordinal.getExtendedPositions &&
+                        ordinal.getExtendedPositions();
+                    ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+                if (!(extendedOrdinalPositions && extendedOrdinalPositions.length)) {
 
-                // first look for an exact match in the ordinalpositions array
-                i = ordinalLength;
-                while (i--) {
-                    if (ordinalPositions[i] === val) {
-                        ordinalIndex = i;
-                        break;
-                    }
+                    return val;
                 }
 
-                // if that failed, find the intermediate position between the
-                // two nearest values
-                i = ordinalLength - 1;
-                while (i--) {
-                    if (val > ordinalPositions[i] || i === 0) { // interpolate
-                        // something between 0 and 1
-                        distance = (val - ordinalPositions[i]) /
-                            (ordinalPositions[i + 1] - ordinalPositions[i]);
-                        ordinalIndex = i + distance;
-                        break;
+                let EOPlength = extendedOrdinalPositions.length,
+                    extendedOrdinalIndex;
+
+                if (!slope) {
+                    slope =
+                        (extendedOrdinalPositions[EOPlength - 1] -
+                            extendedOrdinalPositions[0]) /
+                        EOPlength;
+                }
+                // this value represents the value, where the first index of
+                // ordinalPositions lays in extendedOrdinalPositions.
+                // in other words, how much the OP is moved inside the EOP
```

Comment:
I would unify the names here (`EOP`, `ExtendedOrdinalPositions`, `extendedOrdinalPositions`)
IMO the `rxtendedOrdinalPositions` is the best.